### PR TITLE
feat: repair dependencies in item encryptor

### PR DESCRIPTION
**Pull request by ajewellamz**
*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: DynamoDbEncryption/dafny/DynamoDbEncryptionTransforms/Model/AwsCryptographyDbEncryptionSdkDynamoDbTransformsTypes.dfy
Comment (texastony) [inline DynamoDbEncryption/dafny/DynamoDbEncryptionTransforms/Model/AwsCryptographyDbEncryptionSdkDynamoDbTransformsTypes.dfy:1]: Question: The changes in this file were all generated by Smithy-Dafny, right?
There are no manual changes?

**Review by texastony — COMMENTED** [review]
I have questions and issues I'd like to chat with you about

File: DynamoDbEncryption/dafny/DynamoDbEncryptionTransforms/Model/AwsCryptographyDbEncryptionSdkDynamoDbTransformsTypes.dfy
Comment (ajewellamz) [reply DynamoDbEncryption/dafny/DynamoDbEncryptionTransforms/Model/AwsCryptographyDbEncryptionSdkDynamoDbTransformsTypes.dfy:1]: Correct.

File: DynamoDbEncryption/dafny/DynamoDbEncryptionTransforms/Model/DynamoDbEncryptionTransforms.smithy [hunk 1]
```diff
@@ -96,8 +99,15 @@ structure DynamoDbItemEncryptorReference {}
 @aws.polymorph#reference(service: aws.cryptography.dbEncryptionSdk.dynamoDb#DynamoDbEncryption)
 structure DynamoDbEncryptionReference {}
 
+@aws.polymorph#reference(service: aws.cryptography.dbEncryptionSdk.structuredEncryption#StructuredEncryption)
+structure StructuredEncryptionReference {}
+
+@aws.polymorph#reference(service: aws.cryptography.materialProviders#AwsCryptographicMaterialProviders)
+structure AwsCryptographicMaterialProvidersReference {}
```
[the remark this answers is not in the document]
Comment (texastony) [reply DynamoDbEncryption/dafny/DynamoDbEncryptionTransforms/Model/DynamoDbEncryptionTransforms.smithy:106]: Resolving this and will Approve the PR.

We SHOULD create a GHI against Smithy-Dafny for this,
describing the problem (dependency error discovery when generating Dafny)
the solution (use LocalService's dependencies field for error and include discovery)
and the work around (add references to the local model).

**Review by texastony — APPROVED** [review]

**Pull request merged**